Executble tutorial proposal (updated proposal!)

joseflindkvist opened the pull request:
# Assignment Proposal

## Title

Simulating Serverless Architectures with LocalStack

## Names and KTH ID

- Josef Lindkvist (joslindk@kth.se)
- Josephine Kuo (jkuo@kth.se)

## Deadline

- Task 3

## Category

- Executable Tutorial

## Description

In this executable tutorial, we will demonstrate how to effectively set up LocalStack, a powerful cloud service emulator that allows developers to run and test AWS lambda applications locally, without needing an AWS account. LocalStack runs within a single container, enabling the simulation of AWS services without needing to connect to a remote cloud environment. This setup allows you to fully develop, test, and validate AWS applications locally, enhancing speed and flexibility in your development workflow. Using Killer Coda, we provide an interactive, hands-on experience, ensuring you can explore LocalStack’s capabilities efficiently.


**Relevance**
LocalStack is important within the field of Infrastructure as Code (IaC) as it offers a reliable and efficient way to simulate and validate cloud infrastructure locally. By emulating AWS services, LocalStack accelerates deployment processes, eliminates the need for managing permissions to real cloud accounts, and allows developers to work on different branches without impacting shared infrastructure. This makes it easy to isolate and test changes independently. Moreover, LocalStack enables developers to load complete stacks with all their configurations and data in seconds, bypassing the wait times associated with deploying resources to the cloud, leading to faster iteration cycles and improved development efficiency.

joseflindkvist committed:
Executable tutorial project proposal, Playwright testing

This is a project proposal for an executable tutorial by Josef Lindkvist
and Tobias Ljunggren.

joseflindkvist committed:
End-to-end testing for webapps using Playwright

- Tobias Ljunggren (tljun@kth.se)
- Josef Lindkvist (joslindk@kth.se)

- Task 1, Deadline 1 for async tasks: Oct 1 2024, 17h Stockholm time

- Executable Tutorial

Our executable tutorial will include end-to-end testing for webapps utilizing Playwright (https://playwright.dev/). It will be an executable demonstration on how you can use Playwright in different ways to enhance your end-to-end tests.
We will upload this tutorial to Google Colab.

_Relevance_

This tool is an end-to-end frontend testing framework that is widely
used in the industry. We believe that this tutorial could be useful for
CS students and others who are not very versed in software testing tools
for the web.

joseflindkvist committed:
[200~# Assignment Proposal

End-to-end testing for webapps using Playwright

- Tobias Ljunggren (tljun@kth.se)
- Josef Lindkvist (joslindk@kth.se)

- Week 4

- Demo

Our demo will include end-to-end testing for web apps utilizing
Playwright (https://playwright.dev/). It will demonstrate a couple of
smart ways to use Playwright to enhance your end-to-end tests. In our
demo, we will start by showing how quickly and easily you can get
started with Playwright. We will then live-write a few frontend tests in
a pre-prepared repository. By running the tests, we will show how
Playwright works and utilizes cross-browser testing. Finally, we plan to
demonstrate how you can add Playwright testing to your CI pipeline with
GitHub Actions.

Plan B:
If something doesn't work during our demo we will have screen recordings
prepared to show.

This tool is an end-to-end frontend testing framework that is widely
used in the industry. We believe that this tutorial could be useful for
CS students and others who are not very versed in software testing tools
for the web.

End-to-end testing for webapps using Playwright

- Tobias Ljunggren (tljun@kth.se)
- Josef Lindkvist (joslindk@kth.se)

- Week 4

- Demo

Our demo will include end-to-end testing for web apps utilizing Playwright (https://playwright.dev/). It will demonstrate a couple of smart ways to use Playwright to enhance your end-to-end tests. In our demo, we will start by showing how quickly and easily you can get started with Playwright. We will then live-write a few frontend tests in a pre-prepared repository. By running the tests, we will show how Playwright works and utilizes cross-browser testing. Finally, we plan to demonstrate how you can add Playwright testing to your CI pipeline with GitHub Actions.

Plan B:
If something doesn't work during our demo we will have screen recordings prepared to show.

This tool is an end-to-end frontend testing framework that is widely used in the industry. We believe that this tutorial could be useful for CS students and others who are not very versed in software testing tools for the web.

joseflindkvist committed:
Merge branch '2024' of https://github.com/joseflindkvist/devops-course into 2024

joseflindkvist committed:
changed name of folder

joseflindkvist committed:
formatting

joseflindkvist committed:
new proposal

joseflindkvist committed:
Merge branch 'KTH:2024' into 2024

joseflindkvist committed:
new week proposal

joseflindkvist committed:
new week proposal

joseflindkvist committed:
Merge branch '2024' of https://github.com/joseflindkvist/devops-course into 2024

joseflindkvist committed:
Merge branch 'KTH:2024' into 2024

joseflindkvist committed:
Merge branch 'KTH:2024' into 2024

joseflindkvist committed:
Create a Spotify playlist with Terraform

- Josef Lindkvist (joslindk@kth.se)
- Josephine Kuo (jkuo@kth.se)

- Task 3

- Executable Tutorial

In this executable tutorial, we will showcase how to utilize Terraform. Terraform manages infrastructure on cloud computing providers by building, changing, and versioning infrastructure. In this tutorial, we will create a Spotify playlist using Terraform. The focus will be on Infrastructure as Code principles, showcasing how Terraform can manage and provision cloud infrastructure. We'll use Docker to contain the environment, and Google Colab to execute the code.

**Relevance**
This tutorial is very relevant to the topic of “infrastructure as code” and thus DevOps because it showcases how Terraform can help us define, control, and automate infrastructure. Spotify is a good example of this because its API offers a well-structured, real-world service that requires authentication, permissions, and interactions with an external platform. This mirrors the tasks that DevOps engineers have to perform where automation and provisioning must interact with third-party services.

joseflindkvist committed:
Create a Spotify playlist with Terraform

javierron commented:
@joseflindkvist Hi, thanks for the proposal.
This is very similar to https://github.com/KTH/devops-course/pull/2531, we cannot accept this topic.

joseflindkvist committed:
Merge branch 'KTH:2024' into 2024

joseflindkvist committed:
Update README.md

# Assignment Proposal

## Title

Simulating Serverless Architectures with LocalStack

## Names and KTH ID

- Josef Lindkvist (joslindk@kth.se)
- Josephine Kuo (jkuo@kth.se)

## Deadline

- Task 3

## Category

- Executable Tutorial

## Description

In this executable tutorial, we will demonstrate how to effectively set up LocalStack, a powerful cloud service emulator that allows developers to run and test AWS lambda applications locally, without needing an AWS account. LocalStack runs within a single container, enabling the simulation of AWS services without needing to connect to a remote cloud environment. This setup allows you to fully develop, test, and validate AWS applications locally, enhancing speed and flexibility in your development workflow. Using Killer Coda, we provide an interactive, hands-on experience, ensuring you can explore LocalStack’s capabilities efficiently.


**Relevance**
LocalStack is important within the field of Infrastructure as Code (IaC) as it offers a reliable and efficient way to simulate and validate cloud infrastructure locally. By emulating AWS services, LocalStack accelerates deployment processes, eliminates the need for managing permissions to real cloud accounts, and allows developers to work on different branches without impacting shared infrastructure. This makes it easy to isolate and test changes independently. Moreover, LocalStack enables developers to load complete stacks with all their configurations and data in seconds, bypassing the wait times associated with deploying resources to the cloud, leading to faster iteration cycles and improved development efficiency.

joseflindkvist commented:
Hello @javierron, we have updated our proposal with a new suggestion. Please let us know as soon as possible if it looks good or if you have any comments on it so we can get started :)

algomaster99 commented:
@javierron the proposal looks good and it is original now. Merging!
